syne_tune/optimizer/transfer_learning/bounding_box.py

```diff
@@ -0,0 +1,116 @@
+import logging
+from typing import Dict, Callable, Optional
+import pandas as pd
+
+import syne_tune.search_space as sp
+from syne_tune.optimizer.scheduler import TrialScheduler
+from syne_tune.optimizer.transfer_learning import TransferLearningScheduler, TransferLearningTaskEvaluations
+from syne_tune.search_space import Categorical
+
+
+class BoundingBox(TransferLearningScheduler):
+    def __init__(
+            self,
+            scheduler_fun: Callable[[Dict, str, str], TrialScheduler],
+            config_space: Dict,
+            metric: str,
+            transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            mode: Optional[str] = None,
+            num_hyperparameters_per_task: int = 1,
+    ):
+        """
+        Simple baseline that computes a bounding-box of the best candidate found in previous tasks to restrict the
+         search space to only good candidates.
```

Comment:
Could you add a comment how this new bounding box is computed? What happens if there is only a single task, do the HP ranges all collapse to a single value?
Finally, is config_space passed in used at all, or just to say which HPs there are, and what types?

Reply:
I added a comment. If there is a single task, the ranges are computed on the top `num_hyperparameters_per_task` values. The config-space is used to know the types, identify the hyperparameters and the sampling distribution.